### PR TITLE
[SPARK-30600][SQL] Migrate ALTER VIEW SET/UNSET commands to the new framework

### PR DESCRIPTION
**Pull request by imback82**
### What changes were proposed in this pull request?

Use the new framework to resolve the ALTER VIEW SET/UNSET TBLPROPERTIES commands.

This PR also introduces `UnresolvedView` and `UnresolvedViewWithTableExists` to the framework to handle commands with views.

### Why are the changes needed?

This is a part of effort to make the relation lookup behavior consistent: [SPARK-29900](https://issues.apache.org/jira/browse/SPARK-29900).

### Does this PR introduce any user-facing change?

No

### How was this patch tested?

Added new tests.

### CODE REVIEW

**Comment by imback82** [comment]
cc @cloud-fan

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala [hunk 1]
```diff
@@ -3059,14 +3059,14 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
     val properties = visitPropertyKeyValues(ctx.tablePropertyList)
     val cleanedTableProperties = cleanTableProperties(ctx, properties)
     if (ctx.VIEW != null) {
-      AlterViewSetPropertiesStatement(identifier, cleanedTableProperties)
+      AlterTableSetProperties(UnresolvedView(identifier), cleanedTableProperties)
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala]: shouldn't we use `AlterViewSetProperties`?

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala [hunk 2]
```diff
@@ -3082,7 +3082,7 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
 
     val ifExists = ctx.EXISTS != null
     if (ctx.VIEW != null) {
-      AlterViewUnsetPropertiesStatement(identifier, cleanedProperties, ifExists)
+      AlterTableUnsetProperties(UnresolvedView(identifier), cleanedProperties, ifExists)
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala]: ditto

**Comment by cloud-fan** [comment]
I think we shouldn't reuse the `AlterTableXYZ` commands for views. We should create new `AlterViewXYZ` commands, although they don't have physical plans and always fails during analysis.

**Pull request closed without merging**